tag_matching.c

```diff
@@ -0,0 +1,44 @@
+#include "tag_matching.h"
+#include <sys/ioctl.h>
+#include "util.h"
+static struct pollfd * mpi_poll_fd(MPI_Comm comm, int size)
+{
+	static struct pollfd * fds = NULL;
+	if(fds == NULL){
+	fds = calloc(size, sizeof(struct pollfd));
+	for(size_t i=0; i<size; i++ ){
+		fds[i].fd = comm->socket_info.client_fds[i];
+		fds[i].events = POLLIN; 
+	}
+	}
+	return fds;
```

Comment:
Since this is socket specific code, can you please move it to something like backends/socket/socket_tag_matching.c? Also, can you please add _init to the name of this function, since its job is to init the data structure?

Reply:
Also, `fds` is never freed. Please make sure to free it in `nanompi_free_socket_backend`